crates/api/src/trap.rs

```diff
@@ -62,22 +71,45 @@ impl fmt::Display for Trap {
 impl std::error::Error for Trap {}
 
 #[derive(Debug)]
-pub struct FrameInfo;
+pub struct FrameInfo {
+    module_name: Option<String>,
+    func_index: u32,
+}
 
 impl FrameInfo {
     pub fn instance(&self) -> *const Instance {
         unimplemented!("FrameInfo::instance");
     }
 
-    pub fn func_index() -> usize {
-        unimplemented!("FrameInfo::func_index");
+    pub fn func_index(&self) -> usize {
```

Comment:
Should the return value here be `u32`? (since I think this is indexing into the wasm module namespace?)

Reply:
Also it looks like there's a sentinel value meaning "could not compute", so could this perhaps return an `Option`?